sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastColumnarToRowExec.scala

```diff
@@ -0,0 +1,204 @@
+
+  import GpuMetric._
+  // We need to do this so the assertions don't fail
+  override def supportsColumnar = false
+
+  override def output: Seq[Attribute] = child.output
+
+  override def outputPartitioning: Partitioning = child.outputPartitioning
+
+  override def outputOrdering: Seq[SortOrder] = child.outputOrdering
+
+  // Override the original metrics to remove NUM_OUTPUT_BATCHES, which makes no sense.
+  override lazy val allMetrics: Map[String, GpuMetric] = Map(
+    "dataSize" -> createSizeMetric(ESSENTIAL_LEVEL, "data size"),
+    NUM_OUTPUT_ROWS -> createMetric(outputRowsLevel, DESCRIPTION_NUM_OUTPUT_ROWS),
+    BUILD_TIME -> createNanoTimingMetric(MODERATE_LEVEL, DESCRIPTION_BUILD_TIME),
+    "broadcastTime" -> createNanoTimingMetric(ESSENTIAL_LEVEL, "time to broadcast"),
+    COLLECT_TIME -> createNanoTimingMetric(MODERATE_LEVEL, DESCRIPTION_COLLECT_TIME),
+    TOTAL_TIME -> createNanoTimingMetric(MODERATE_LEVEL, DESCRIPTION_TOTAL_TIME),
+    NUM_INPUT_BATCHES -> createMetric(DEBUG_LEVEL, DESCRIPTION_NUM_INPUT_BATCHES))
+
+  @transient
+  private lazy val promise = Promise[Broadcast[Any]]()
+
+  @transient
+  private val timeout: Long = SQLConf.get.broadcastTimeout
+
+  /**
+   * For registering callbacks on `relationFuture`.
+   * Note that calling this field will not start the execution of broadcast job.
+   */
+  @transient
+  lazy val completionFuture: concurrent.Future[Broadcast[Any]] = promise.future
+
+  val _runId: UUID = UUID.randomUUID()
+
+  @transient
+  lazy val relationFuture: Future[Broadcast[Any]] = {
+    // relationFuture is used in "doExecute". Therefore we can get the execution id correctly here.
+    val executionId = sparkContext.getLocalProperty(SQLExecution.EXECUTION_ID_KEY)
+    val numOutputRows = gpuLongMetric(NUM_OUTPUT_ROWS)
+    val collectTime = gpuLongMetric(COLLECT_TIME)
+    val buildTime = gpuLongMetric(BUILD_TIME)
+    val broadcastTime = gpuLongMetric(BUILD_TIME)
+    val totalTime = gpuLongMetric(TOTAL_TIME)
+
+    val task = new Callable[Broadcast[Any]]() {
+      override def call(): Broadcast[Any] = {
+        // This will run in another thread. Set the execution id so that we can connect these jobs
+        // with the correct execution.
+        SQLExecution.withExecutionId(sqlContext.sparkSession, executionId) {
+          val totalRange = new MetricRange(totalTime)
+          try {
+            // Setup a job group here so later it may get cancelled by groupId if necessary.
+            sparkContext.setJobGroup(_runId.toString, s"broadcast exchange (runId ${_runId})",
+              interruptOnCancel = true)
+
+            val f = GpuColumnarToRowExecParent
+                .makeIteratorFunc(broadcast.child.output, numOutputRows, NoopMetric, totalTime)
+
+            val collectRange = new NvtxWithMetrics("broadcast collect", NvtxColor.GREEN,
+              collectTime)
+            val serializedBatch: SerializeConcatHostBuffersDeserializeBatch = try {
```

Comment:
I don't think this is going to work. `SerializeConcatHostBuffersDeserializeBatch` will deserialize the data to the GPU.  This code runs on the driver, which is not guaranteed to have a GPU.